encodings/dict/src/ops.rs

```diff
@@ -6,8 +8,14 @@ use crate::DictArray;
 
 impl ArrayOperationsImpl for DictArray {
     fn _slice(&self, start: usize, stop: usize) -> VortexResult<ArrayRef> {
-        DictArray::try_new(self.codes().slice(start, stop)?, self.values().clone())
-            .map(|a| a.into_array())
+        let sliced_code = self.codes().slice(start, stop)?;
+        if sliced_code.is_encoding(ConstantEncoding.id()) {
+            let code = usize::try_from(&sliced_code.scalar_at(0)?)?;
```

Comment:
Looks like the code is coming back as null, meaning it's actually a constant null array!